src/app/in-memory-data.service.ts

```diff
@@ -327,6 +327,89 @@ export class InMemoryDataService implements InMemoryDbService {
       'imageURL': 'https://avatars.githubusercontent.com/u/2410474?v=3'
     };
 
-    return {loginStatus, workitems, workitemtypes, user};
+    let linkCategories = {
+      "meta": { "comment": "Note, that the 'id' field is optional for a create request." },
+      "id": "1",
+      "type": "linkcategories",
+      "attributes": {
+        "name": "test",
+        "title": "This category is reserved for link types that belong to the 'test' subsystem."
+      }
+    }
+
+    let linkTypes = [
+      {
+        "type": "linktypes",
+        "id": "1",
+        "attributes": {
+          "name": "tested-by-link-type",
+          "description": "A test work item can 'test' if a the code in a pull request passes the tests.",
+          "source_type": "test-workitemtype",
+          "target_type": "pull-request-workitemttype",
+          "forward_name": "tests",
+          "reverse_name": "tested by"
+        },
+        "relationships": {
+          "link_category": {
+            "meta": { "comment": "See the creation of link categories for what this ID means" },
+            "data": { "type": "link_category", "id": "6c5610be-30b2-4880-9fec-81e4f8e4fd76" },
+            "links": {
+              "self": "http://example.com/api/linktypes/40bbdd3d-8b5d-4fd6-ac90-7236b669af04/relationships/link_category",
+              "related": "http://example.com/api/linktypes/40bbdd3d-8b5d-4fd6-ac90-7236b669af04/link_category"
+            }
+          }
+        },
+        "links": {
+          "self": "http://example.com/api/linktypes/40bbdd3d-8b5d-4fd6-ac90-7236b669af04"
+        }
+      },
+      {
+        "type": "linktypes",
+        "id": "2",
+        "attributes": {
+          "name": "tested-by-link-type",
+          "description": "A test work item can 'test' if a the code in a pull request passes the tests.",
+          "source_type": "test-workitemtype",
+          "target_type": "pull-request-workitemttype",
+          "forward_name": "tests1",
+          "reverse_name": "tested by"
+        },
+        "relationships": {
+          "link_category": {
+            "meta": { "comment": "See the creation of link categories for what this ID means" },
+            "data": { "type": "link_category", "id": "6c5610be-30b2-4880-9fec-81e4f8e4fd76" },
+            "links": {
+              "self": "http://example.com/api/linktypes/40bbdd3d-8b5d-4fd6-ac90-7236b669af04/relationships/link_category",
+              "related": "http://example.com/api/linktypes/40bbdd3d-8b5d-4fd6-ac90-7236b669af04/link_category"
+            }
+          }
+        },
+        "links": {
+          "self": "http://example.com/api/linktypes/40bbdd3d-8b5d-4fd6-ac90-7236b669af04"
+        }
+      }
+    ];
+
+    let link = {
```

Comment:
This is how a link looks like now. See also http://swagger.goa.design/?url=kwk%2Falmighty-core%2Fdesign#!/work-item-link/work_item_link_show

Notice that except for `version`, there's no attribute any longer. The `relationships` object hold the major data.

```json
{
  "data": {
    "attributes": {
      "version": 0
    },
    "id": "40bbdd3d-8b5d-4fd6-ac90-7236b669af04",
    "relationships": {
      "link_type": {
        "data": {
          "id": "6c5610be-30b2-4880-9fec-81e4f8e4fd76",
          "type": "workitemlinktypes"
        }
      },
      "source": {
        "data": {
          "id": "1234",
          "type": "workitems"
        }
      },
      "target": {
        "data": {
          "id": "1234",
          "type": "workitems"
        }
      }
    },
    "type": "workitemlinks"
  }
}
```